src/Masa.Utils.Exceptions/Handlers/ExceptionHandlerMiddleware.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Utils.Exceptions.Handlers;
+
+public class ExceptionHandlerMiddleware
+{
+    private readonly RequestDelegate _next;
+    private readonly MasaExceptionHandlerOptions _options;
+    private readonly MasaExceptionLogRelationOptions _logRelationOptions;
+    private readonly ILogger<ExceptionHandlerMiddleware>? _logger;
+
+    public ExceptionHandlerMiddleware(
+        RequestDelegate next,
+        IOptions<MasaExceptionHandlerOptions> options,
+        IOptions<MasaExceptionLogRelationOptions> logRelationOptions,
+        ILogger<ExceptionHandlerMiddleware>? logger = null)
+    {
+        _next = next;
+        _options = options.Value;
+        _logRelationOptions = logRelationOptions.Value;
+        _logger = logger;
+    }
+
+    public async Task InvokeAsync(HttpContext httpContext)
+    {
+        try
+        {
+            await _next(httpContext);
+        }
+        catch (Exception exception)
+        {
+            var masaExceptionContext = new MasaExceptionContext(exception, httpContext);
+            _options.ExceptionHandler?.Invoke(masaExceptionContext);
+
+            if (httpContext.Response.HasStarted)
+                return;
+
+            if (masaExceptionContext.ExceptionHandled)
+            {
+                await httpContext.Response.WriteTextAsync(
+                    masaExceptionContext.StatusCode,
+                    masaExceptionContext.Message ?? masaExceptionContext.Exception.Message,
+                    masaExceptionContext.ContentType);
+                return;
+            }
+
+            _logger?.WriteLog(masaExceptionContext.Exception, LogLevel.Error, _logRelationOptions);
```

Comment:
If the exception is UserFriendlyException, can it be Error?

Reply:
The exception of UserFriendlyException does record Error before, I can't confirm this, as I think, it is `Warning` at most

Reply:
Mmmm now the new one is Information